### PR TITLE
pulp: remove pulp and all references to it

### PR DESCRIPTION
**Pull request by mlangsdorf**
fixes osbs5098

Remove pulp and all references to it from atomic-reactor.  Delete the 3 remaining pulp plugins, delete_from_registry plugin, and pulp utils, along with all related test code.

Signed-off-by: Mark Langsdorf <mlangsdo@redhat.com>

DO NOT MERGE - this is the master commit for removing pulp, and is too large to commit.  Separate PRs will unwind the pulp infrastructure.

- [x] Remove the pulp plugins and delete_from_registry, leaving their plugin keys and pulp_registries in push_conf.  See #1273 
- [x] Remove the plugin keys and references to them, but leave pulp_registries in push_conf.  See #1278 
- [ ] Remove pulp_registries.

Maintainers will complete the following section:
- [ ] Commit messages are descriptive enough
- [ ] "Signed-off-by:" line is present in each commit
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] JSON/YAML configuration changes are updated in the relevant schema
- [ ] Changes to metadata also update the documentation for the metadata
- [ ] Pull request includes link to an osbs-docs PR for user documentation updates
- [ ] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/exit_koji_import.py [hunk 1]
```diff
@@ -213,9 +178,8 @@ def set_media_types(self, extra, worker_metadatas):
                 media_types = json.loads(annotations['media-types'])
                 break
 
-        # Append media_types from pulp pull or verify images
-        media_results = (self.workflow.exit_results.get(PLUGIN_PULP_PULL_KEY) or
-                         self.workflow.exit_results.get(PLUGIN_VERIFY_MEDIA_KEY))
+        # Append media_types from verify images
+        media_results = (self.workflow.exit_results.get(PLUGIN_VERIFY_MEDIA_KEY))
```
Comment (twaugh) [inline atomic_reactor/plugins/exit_koji_import.py]: No longer need parentheses.

File: atomic_reactor/plugins/exit_store_metadata_in_osv3.py [hunk 2]
```diff
@@ -262,11 +251,7 @@ def run(self):
 
         media_types = []
 
-        # pulp_pull may run on worker as a postbuild plugin or on orchestrator as an exit plugin
-        # verify_media_results runs if pulp_pull does not
-        media_results = (self.workflow.postbuild_results.get(PLUGIN_PULP_PULL_KEY) or
-                         self.workflow.exit_results.get(PLUGIN_PULP_PULL_KEY) or
-                         self.workflow.exit_results.get(PLUGIN_VERIFY_MEDIA_KEY))
+        media_results = (self.workflow.exit_results.get(PLUGIN_VERIFY_MEDIA_KEY))
```
Comment (twaugh) [inline atomic_reactor/plugins/exit_store_metadata_in_osv3.py]: No longer need parentheses.

File: tests/plugins/test_koji_import.py [hunk 3]
```diff
@@ -1855,11 +1787,9 @@ def test_koji_import_set_manifest_list_info(self, caplog, tmpdir, monkeypatch, o
...
 
-    @pytest.mark.skipif(not PULP_SYNC_AVAILABLE,
-                        reason="pulp_sync not available")
     @pytest.mark.parametrize('available,expected', [
         (None, ['sha256:v1', 'sha256:v2']),
-        (['foo', 'sha256:v1'], ['sha256:v1']),
+        (['foo', 'sha256:v1'], ['sha256:v1', 'sha256:v2']),
         (['sha256:v1', 'sha256:v2'], ['sha256:v1', 'sha256:v2']),
     ])
     def test_koji_import_unavailable_manifest_digests(self, tmpdir, os_env,
```
Comment (twaugh) [inline tests/plugins/test_koji_import.py]: I think this whole test should go. We're no longer using `available`, and without that parameter I'm not sure what it's testing.

File: tests/plugins/test_store_metadata.py [hunk 4]
```diff
@@ -90,15 +87,10 @@ def df_path(self):
...
+def prepare(docker_registries=None, before_dockerfile=False,  # noqa
             reactor_config_map=False):
-    if pulp_registries is None:
-        pulp_registries = (
-            ("test", LOCALHOST_REGISTRY),
-        )
-
     if docker_registries is None:
-        docker_registries = (DOCKER0_REGISTRY,)
+        docker_registries = (LOCALHOST_REGISTRY, DOCKER0_REGISTRY,)
```
Comment (twaugh) [inline tests/plugins/test_store_metadata.py]: That's a bit unexpected. Is that intentional? We weren't doing it before.

Comment (mlangsdorf) [reply tests/plugins/test_store_metadata.py]: the `pulp_registries` argument is effectively None, except the argument is getting removed.
Previously, when the `pulp_registries` argument was None, pulp_registries was set to ("test", LOCALHOST_REGISTRY) and that was added to push_conf.pulp_registries and push_conf.all_registries.  This is the minimal change to preserve that behavior.

The other option is to remove all of the LOCALHOST_REGISTRY entries from expected.  That's probably more correct, so I'll go ahead and do that.

File: tests/plugins/test_koji_import.py (hunk 3 above)
Comment (mlangsdorf) [reply tests/plugins/test_koji_import.py]: Fixed.

File: atomic_reactor/plugins/exit_store_metadata_in_osv3.py (hunk 2 above)
Comment (mlangsdorf) [reply atomic_reactor/plugins/exit_store_metadata_in_osv3.py]: Fixed.

File: atomic_reactor/plugins/exit_koji_import.py (hunk 1 above)
Comment (mlangsdorf) [reply atomic_reactor/plugins/exit_koji_import.py]: Fixed.

**Pull request closed without merging**